src/thrift-client/TgtInterfaceImpl.cpp

```diff
@@ -848,6 +844,88 @@ void StordVmdk::UpdateStats(Request* reqp) {
 		} else {
 			stats_.truncate_latency_ += latency;
 		}
+	case Request::Type::kSync:
+		/* No stats requires for sync as it will not fails in thrift */
+		break;
+	}
+}
+
+std::ostream& operator << (std::ostream& os, const RequestBase::Type type) {
+	switch (type) {
+	case RequestBase::Type::kRead:
+		os << "read";
+		break;
+	case RequestBase::Type::kWrite:
+		os << "write";
+		break;
+	case RequestBase::Type::kWriteSame:
+		os << "writesame";
+		break;
+	case RequestBase::Type::kTruncate:
+		os << "truncate";
+		break;
+	case RequestBase::Type::kSync:
+		os << "sync";
+		break;
+	}
+	return os;
+}
+
+std::ostream& operator << (std::ostream& os, const SyncRequest& request) {
+	os << "ID " << request.id
+		<< " type " << request.type
+		<< " privatep " << request.privatep
+		<< " offset " << request.offset
+		<< " length " << request.length;
+	return os;
+}
+
+std::ostream& operator << (std::ostream& os, const Request& request) {
+	os << "ID " << request.id
+		<< " type " << request.type
+		<< " privatep " << request.privatep
+		<< " bufferp " << request.bufferp
+		<< " buf_sz " << request.buf_sz
+		<< " offset " << request.offset
+		<< " length " << request.length;
+	return os;
+}
+
+RequestBase::RequestBase(RequestID id, Type t, const void* privatep,
+	uint64_t length, int64_t offset) : id(id), type(t), privatep(privatep),
+	length(length), offset(offset) {
+}
+
+RequestBase::~RequestBase() {
+}
+
+const RequestBase::Type& RequestBase::GetType() const noexcept {
+		return type;
+}
+
+bool RequestBase::IsOverlapped(uint64_t req_offset,
+	uint64_t req_length) const noexcept {
+	return ((int64_t)(length - req_offset) > 0 &&
+		(int64_t)(req_length - offset) > 0);
```

Comment:
@dmehta85 this condition is wrong. Please test this code in isolation.

Reply:
Typecasting uint64_t to int64_t should also be avoided.

Reply:
Based on the section 'How to check writes are overlapped' given in the [design document](https://primaryio.atlassian.net/secure/attachment/12388/SyncDesign.pdf), I have written an isolated program and found a condition works perfectly. 
[Isolated Program](https://primaryio.atlassian.net/secure/attachment/12397/check_overlap.c)  uploaded on jira ticket [DP-2124](https://primaryio.atlassian.net/browse/DP-2124)

Program output:
#> ./check_overlap 
Write[1] is overlapped: No
Write[2] is overlapped: Yes
Write[3] is overlapped: Yes
Write[4] is overlapped: Yes
Write[5] is overlapped: Yes
Write[6] is overlapped: No
Write[7] is overlapped: Yes
#>

Please share the condition in which above logic becomes false.

Reply:
1. Assume, existing request  has Offset = 1M (1048576), Length = 4096
2. New request is req_offset = 1M + 512 = 1049088 and Length = 512

These are overlapping requests 

1. length - req_offset > 0 = 4096 - (1M+512)  > 0 = false
2. req_length - offset > 0 = 512 - (1M) > 0 = false

Your program checks with pretty small numbers.

Reply:
Calculation is corrected.